src/main/kotlin/com/jakewharton/dex/DexMethods.kt

```diff
@@ -18,20 +21,37 @@ import java.util.zip.ZipInputStream
 
 /** Extract method references from dex bytecode. */
 class DexMethods private constructor() {
+  /** Configuration flags populated at startup. */
+  private class Config {
+    @Option(name="--hide-synthetic-numbers")
+    var hideSyntheticNumbers: Boolean = false
+
+    @Option(name="--proguard-map")
+    var proguardMap : String? = null
+
+    @Argument
+    var files : List<String> = ArrayList()
+  }
+
   companion object {
     private val CLASS_MAGIC = byteArrayOf(0xCA.toByte(), 0xFE.toByte(), 0xBA.toByte(), 0xBE.toByte())
     private val DEX_MAGIC = byteArrayOf(0x64, 0x65, 0x78, 0x0a, 0x30, 0x33, 0x35, 0x00)
     private val SYNTHETIC_SUFFIX = ".*?\\$\\d+".toRegex()
 
     @JvmStatic fun main(vararg args: String) {
-      val hideSyntheticNumbers = args.contains("--hide-synthetic-numbers")
-      val bytesList = args
-          .filter { !it.startsWith("--") }
+      val config = Config()
+      CmdLineParser(config).parseArgument(args.toList())
+
+      val proguardMap = config.proguardMap
+      val methodMapper : ((MethodName) -> MethodName) =
+          if (proguardMap != null) ProguardNameMapper(proguardMap) else ({ it })
+
+      val bytesList = config.files
           .map { FileInputStream(it) }
           .defaultIfEmpty(System.`in`)
-          .map { it.readBytes() }
+          .map { it.use { it.readBytes() } }
```

Comment:
Thanks :) If you'd like it as a separate PR (since it's much smaller) I can send it out